test/controllers/tasks_controller_test.rb

```diff
@@ -88,37 +82,91 @@
   # Unskip and complete these tests for Wave 3
   describe "edit" do
     it "can get the edit page for an existing task" do
-      skip
-      # Your code here
+      get edit_task_path(task.id)
+      must_respond_with :success
     end
     
     it "will respond with redirect when attempting to edit a nonexistant task" do
-      skip
-      # Your code here
+      get edit_task_path(-1)
+      must_respond_with :redirect
     end
   end
   
   # Uncomment and complete these tests for Wave 3
   describe "update" do
-    # Note:  If there was a way to fail to save the changes to a task, that would be a great
-    #        thing to test.
+    before do
+      @task_hash = {
+          task: {
+              name: task[:name],
+              description: "Describe your task",
+              completed_at: task[:completed_at],
+          }
+      }
+    end
     it "can update an existing task" do
-      # Your code here
+      expect {
+        patch task_path(task.id), params: @task_hash
+      }.wont_change "Task.count"
+
+      new_task = Task.find_by(name: @task_hash[:task][:name])
+      expect(new_task.description).must_equal @task_hash[:task][:description]
+
+      must_respond_with :redirect
+      must_redirect_to task_path(task.id)
     end
     
     it "will redirect to the root page if given an invalid id" do
-      # Your code here
+      expect {
+        patch task_path(-1), params: @task_hash
+      }.wont_change "Task.count"
+
+      must_redirect_to root_path
     end
   end
-  
-  # Complete these tests for Wave 4
-  describe "destroy" do
-    # Your tests go here
-    
+
+  describe 'destroy' do
+    it 'will destroy to permanently remove a task' do
+      task
+      expect {
+        destroy task_path(task.id)
```

Comment:
This test and the test below are failing because the proper HTTP verb is "delete" rather than "destroy".
```suggestion
        delete task_path(task.id)
```

Reply:
Hence I get the following error
```
NoMethodError: undefined method `destroy'
```